### ISSUE TITLE
How can I annotate my own dataset?

### ISSUE THREAD

**Issue by Xuguozi** [TencentARC/UMT#11]
I am very interested in your research. I have a lot of questions, thank you for taking the time to answer!

1. How to annotate the .json file using my own dataset?

2. How to annotate the label in the video? What is the meaning of match in the .json annotation file, I read from the reference " Label 1 denotes matched clip, label -1 denotes unmatched clip, and label 0 denotes borderline cases.". What does the 0 boundary mean? Does the clip with match 0 contain both positive and negative samples? How do I tag my videos?

3. How to segment the video. I refer to youtube's .json annotation file, it shows that a clip contains 100 frames, and the overlap is 50%. But I saw in the paper that "Since each feature vector captures 32 consecutive frames, we follow and consider the feature vector belonging to a clip if their overlap is more than 50%." What is the relationship between 32 frames and 100 frames? How many frames should I use for a clip?

4. My lab questions. My dataset is each 5s video segment (train: 1554, val: 389.), and each video contains only positive or negative samples. Each clip is 50 frames, and the overlap is 50%. But my result is loss change, mAP and best mAP do not change (as shown in the figure), what is the reason?
1) Is it related to overfitting? 2) Is it related to the division of frames? 3) Does it have something to do with the data only containing a certain sample? Or the video segment should have both positive and negative samples. If a video segment contains both positive and negative samples, how should I label the video?

![1](https://user-images.githubusercontent.com/89565268/170275283-ebc9ecfb-6fa0-4a7c-90f1-cd9a73901e03.png)

![2](https://user-images.githubusercontent.com/89565268/170275385-35eec6d4-e734-4cba-9138-94ba1a9c611a.png)

5. Where can I download the raw video data of YouTube Highlight?

Thank you very much!

**Comment by yeliudev** [comment]
1. You may refer to YouTube Highlights's [repo](https://github.com/aliensunmin/DomainSpecificHighlight) about the annotations. We used `match` scores as the highlight annotations.
2. We regarded all the `0` labels as negative samples.
3. Each clip in I3D contains 32 frames, while each clip in the annotations contains 100 frames. Thus you need to map the annotations from 100 frames/clip to 32 frames/clip.
4. You may double-check your annotations. If a clip contains both positive and negative frames, you can label it using the label that most of the frames have.
5. Duplicated #12.
